src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1887,6 +1888,155 @@ public static void XmlSerializerFactoryTest()
         Assert.Equal(dog1.Breed, dog2.Breed);
     }
 
+    static List<string> grouplists;
+    static AutoResetEvent resetevent = new AutoResetEvent(false);
+    static int count = 0;
+    [Fact]
+    public static void XmlUnknownElementAndEventHandlerTest()
+    {
+        XmlSerializer serializer = new XmlSerializer(typeof(Group));
+        serializer.UnknownElement += new XmlElementEventHandler((o, args) =>
+        {
+            Group myGroup = (Group)args.ObjectBeingDeserialized;
+            Assert.NotNull(myGroup);
+            grouplists.Add(args.Element.Name);
+            ++count;
+            if (count >= 3) resetevent.Set();
```

Comment:
I don't think we need to use AutoResetEvent. We just need to set a flag in the event handler to verify the handler is called.